python-checks/src/test/resources/checks/duplicatedMethodImplementationCheck.py

```diff
@@ -0,0 +1,160 @@
+class clazz:
+  def method(self):
+    foo()
+    bar()
+
+  def method2(self): #Noncompliant
+#     ^^^^^^^
+    foo()
+    bar()
+
+  def method3(self): #Noncompliant
+#     ^^^^^^^
+    foo()
+    bar()
+
+
+class clazz:
+  def method(self):
+    bar()
+    foo()
+
+  def method2(self):
+    foo()
+    bar()
+
+class clazz:
+  def method(self):
+    'doc'
+    bar()
+    foo()
+
+  def method2(self):
+    'doc'
+    foo()
+    bar()
+
+class clazz:
+  def method(self):
+    if cond:
+      foo()
+    else:
+      bar()
+
+  def method(self): # Noncompliant
+    if cond:
+      foo()
+    else:
+      bar()
+
+class clazz:
+  def method(self):
+    foo()
+
+  def method2(self): # ok, exception for 1 line methods
+    foo()
+
+
+class clazz:
+  def __enter__(self):
+    'doc'
+    pass
+
+  def __exit__(self): # ok, docstring is not counted as a line of code
+    'doc'
+    pass
+
+class clazz:
+  def __enter__(self):
+    'doc'
+
+  def __exit__(self): # ok, docstring is not counted as a line of code
+    'doc'
+
+class clazz:
+  def __enter__(self):
+    '''
+    Some multiline
+    docstring
+    '''
+    pass
+
+  def __exit__(self): # ok, docstring is not counted as a line of code
+    '''
+    Some multiline
+    docstring
+    '''
+    pass
+
+class clazz:
+  def method(self):
+    class nested:
+      def nestedMethod(self):
+        foo()
+        bar()
+
+  def method(self): #ok, not the same class
+    foo()
+    bar()
+
+class clazz:
+  def method(self):
+    def nested():
+      foo()
+      bar()
+
+  def method(self): #ok, nested is not a method
+    foo()
+    bar()
+
+class clazz:
+  def not_implemented_method(self):
+    raise NotImplementedError("Some message")
+
+  def not_implemented_duplicate(self):
+    raise NotImplementedError("Some message")
+
+class clazz:
+  def not_implemented_method(self):
+    'docstringc'
+    raise NotImplementedError("Some message")
+
+  def not_implemented_duplicate(self):
+    'docstringc'
+    raise NotImplementedError("Some message")
+
+class clazz:
+  def not_implemented_method(self):
+    raise NotImplementedError(
+      "Some message"
+    )
+
+  def not_implemented_duplicate(self):
+    raise NotImplementedError(
+      "Some message"
+    )
+
+class clazz:
+  def not_implemented_method_with_docstring(self):
+    '''
+    This is a docstring
+    '''
+    raise NotImplementedError(
+      "Some message"
+    )
+
+  def not_implemented_duplicate(self):
+    '''
+    This is a docstring
+    '''
+    raise NotImplementedError(
+      "Some message"
+    )
+
+#coverage
```

Comment:
What's the meaning of this comment? If the class which is defined below is there to test something specific (one line methods, maybe), let's mention it. It's also possible to drop the comment and rename the class itself, for example to "OneLineMethods".